.github/workflows/build.yml

```diff
@@ -77,7 +77,7 @@ jobs:
           submodules: true
 
       - name: Configure
-        run: cmake -DPCRE2_SUPPORT_JIT=ON -DPCRE2_BUILD_PCRE2_16=ON -DPCRE2_BUILD_PCRE2_32=ON -DCMAKE_OSX_ARCHITECTURES='arm64;x86_64' -DCMAKE_C_FLAGS='-Wall -Wextra' -DCMAKE_COMPILE_WARNING_AS_ERROR=ON -B build
+        run: cmake -DPCRE2_SUPPORT_JIT=ON -DPCRE2_BUILD_PCRE2_16=ON -DPCRE2_BUILD_PCRE2_32=ON -DCMAKE_OSX_ARCHITECTURES='arm64;x86_64' -DCMAKE_C_FLAGS='-Wall -Wextra' -DCMAKE_COMPILE_WARNING_AS_ERROR=ON -DCMAKE_BUILD_TYPE=RelWithDebInfo -B build
```

Comment:
I think all the jobs in `build.yml` should be using RelWithDebInfo (for the jobs using CMake of course). And, we want PCRE2_DEBUG to be left to its default, which will disable the assertions.

This mirrors the configuration which we expect customers and downstream packages to be using.

Reply:
The thinking was that we use the default of the generator used since they are meant to match what a developer runs localy, but you probably have a lot more experience with cmake as a developer than any of us.

Reply:
I never run CMake without specifying a CMAKE_BUILD_TYPE. Some generators (like the Makefiles backend) just do weird things, and generate configuration that nobody really wants.

I guess we could have one example of that in the dev.yml... but it would only apply on Linux anyway, and we're expecting most Linux users to be using autoconf instead. So I'd just simplify things, and only test on actual proper CMake configurations (Debug, Release, RelWithDebInfo).

Reply:
I realize I misread the context.

My comment was bout jobs in dev.yml, for build.yml we want to mirror what end users would do for production, so while I think you are correct `RelWithDebInfo` is valid `Release` might be better (which will also test -O3 vs -O2 with gcc/clang)

Reply:
Interesting. I've personally always considered "RelWithDebInfo" to be the best for actual production releases.

To start with, no-one should be shipping binaries without having the debug info stored somewhere. Linux distributions always build with debug info, and ship the ELF+DWARF binaries in special "-debug" variant of the package, then run `strip` to publish the ordinary packages. That way, when users report a crash, you can symbolise it. Commercial vendors do the same thing (on all the companies and projects I've worked on) - build with symbols, store them, and ship the stripped binaries.

I also (personally) dislike O3 because it does too much inlining with GCC. Binary size for a C++ app can double, with no noticeable performance increase. For apps that are bigger than 10-20MB binary size, `-Os` is often the fastest in fact, because `-O3` does so much loop unrolling it forces the code out of the L1 cache.

But yes, I totally acknowledge that people ship with `-O2`, `-O3`, or `-Os`.

Here's what I'll do: change `build.yml` to use `Release` (`-O3`), and in `build.yml` let's make sure we run the unit tests for each of `-O2`, `-O3`, and `-Os`.